bpf/bpf_overlay.c

```diff
@@ -673,6 +674,21 @@ int cil_from_overlay(struct __ctx_buff *ctx)
 		goto out;
 	}
 
+#if defined(ENABLE_WIREGUARD)
+	/* When wireguard is enabled we should drop any traffic coming through the tunnel
+	 * that previously wasn't marked as decrypted by cilium.
+	 */
+	if (CONFIG(encryption_strict_ingress) && ctx->mark != MARK_MAGIC_DECRYPT) {
+		ret = DROP_INVALID;
```

Comment:
Why not DROP_UNENCRYPTED_TRAFFIC?

Reply:
Must have missed this drop reason, but indeed a better fit! :pray: